### DISCUSSION TITLE
Component testing of Vitest

### DISCUSSION THREAD

**Discussion by edimitchel** [vitest-dev/vitest#572] [General]
For the moment, the UI is not covered by unit test (ironic for a test runner).

In the **ui** discord channel, we talk about using Cypress or Playwright.

We should consider component testing (with mocked API) and E2E test (this would be very satisfying to see a test runner test another test runner).

Let's talk here!

**Comment by JessicaSachs** [comment]
I'm the maintainer of Cy Component Testing, and I'd offer support to integrate and write some component tests.

For e2e, I think either Cy or Playwright would be sufficient. The decision should probably be driven by how easy it is to mock the network.

You 10000% do need both kinds of tests though.

**Comment by edimitchel** [comment]
I just start a thing here: 
- https://github.com/vitest-dev/vitest/pull/585

**Reply by edimitchel** [reply]
@JessicaSachs worked on it:
- #590
